# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsHistory.cpp
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc dom/base/nsHistory.cpp
--- a/dom/base/nsHistory.cpp	Wed Apr 28 11:09:38 2021 +0000
+++ b/dom/base/nsHistory.cpp	Wed Apr 28 11:26:49 2021 +0000
@@ -15,6 +15,7 @@
 #include "nsIWebNavigation.h"
 #include "nsReadableUtils.h"
 #include "nsContentUtils.h"
+#include "mozilla/dom/WindowContext.h"
 #include "mozilla/dom/Location.h"
 #include "mozilla/RefPtr.h"
 #include "mozilla/StaticPrefs_dom.h"
@@ -161,11 +162,17 @@
   // lead to exceptions and a possible leak of history length
   // AsyncGo throws if we hit the location change rate limit.
   if (StaticPrefs::dom_window_history_async()) {
-    session_history->AsyncGo(aDelta, /* aRequireUserInteraction = */ false,
-                             aCallerType, aRv);
+    session_history->AsyncGo(
+        aDelta, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        aCallerType, aRv);
   } else {
-    session_history->Go(aDelta, /* aRequireUserInteraction = */ false,
-                        IgnoreErrors());
+    session_history->Go(
+        aDelta, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        IgnoreErrors());
   }
 }
 
@@ -185,10 +192,17 @@
   }
 
   if (StaticPrefs::dom_window_history_async()) {
-    sHistory->AsyncGo(-1, /* aRequireUserInteraction = */ false, aCallerType,
-                      aRv);
+    sHistory->AsyncGo(
+        -1, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        aCallerType, aRv);
   } else {
-    sHistory->Go(-1, /* aRequireUserInteraction = */ false, IgnoreErrors());
+    sHistory->Go(
+        -1, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        IgnoreErrors());
   }
 }
 
@@ -208,10 +222,17 @@
   }
 
   if (StaticPrefs::dom_window_history_async()) {
-    sHistory->AsyncGo(1, /* aRequireUserInteraction = */ false, aCallerType,
-                      aRv);
+    sHistory->AsyncGo(
+        1, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        aCallerType, aRv);
   } else {
-    sHistory->Go(1, /* aRequireUserInteraction = */ false, IgnoreErrors());
+    sHistory->Go(
+        1, /* aRequireUserInteraction = */ false,
+        /* aUserActivation */
+        win->GetWindowContext()->HasValidTransientUserGestureActivation(),
+        IgnoreErrors());
   }
 }
 